sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/workflow/whenphase/routeinvoker/impl/KafkaRouteInvoker.java

```diff
@@ -0,0 +1,134 @@
+  private final List<String> kafkaSpecificHeaderKeys =
+      List.of(
+          PARTITION_KEY,
+          PARTITION,
+          KEY,
+          TOPIC,
+          OVERRIDE_TOPIC,
+          OFFSET,
+          HEADERS,
+          LAST_RECORD_BEFORE_COMMIT,
+          LAST_POLL_RECORD,
+          TIMESTAMP,
+          OVERRIDE_TIMESTAMP);
+
+  @Override
+  public Optional<Exchange> invoke(Exchange inputExchange) {
+    KafkaConsumer kafkaConsumer = (KafkaConsumer) resolveConsumer(inputExchange, camelContext);
+
+    Exchange kafkaExchange = kafkaConsumer.createExchange(false);
+    kafkaExchange.getMessage().setBody(inputExchange.getMessage().getBody());
+    kafkaExchange
+        .getMessage()
+        .setHeaders(
+            createExchangeHeaders(
+                inputExchange.getMessage().getHeaders(),
+                kafkaConsumer.getEndpoint().getConfiguration()));
+
+    try {
+      kafkaConsumer.getProcessor().process(kafkaExchange);
+    } catch (Exception e) {
+      log.error("sip.testkit.workflow.whenphase.routeinvoker.kafka.badrequest");
+    }
+
+    return Optional.empty();
+  }
+
+  @Override
+  public boolean isApplicable(Endpoint endpoint) {
+    return endpoint instanceof KafkaEndpoint;
+  }
+
+  @Override
+  public boolean shouldSuspend(Endpoint endpoint) {
+    return true;
+  }
+
+  private Map<String, Object> createExchangeHeaders(
+      Map<String, Object> inputHeaders, KafkaConfiguration configuration) {
+    Map<String, Object> kafkaHeaders =
+        prepareKafkaSpecificHeaders(inputHeaders, configuration.getTopic());
+    kafkaHeaders.putAll(prepareCustomHeaders(inputHeaders));
+    return kafkaHeaders;
+  }
+
+  private Map<String, Object> prepareKafkaSpecificHeaders(
+      Map<String, Object> inputHeaders, String kafkaTopic) {
+    Map<String, Object> kafkaSpecificHeaders =
+        inputHeaders.entrySet().stream()
+            .filter(header -> isKafkaSpecificHeader(header.getKey()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    kafkaSpecificHeaders.putIfAbsent(TOPIC, kafkaTopic);
+    long currentTimeMillis = System.currentTimeMillis();
+    kafkaSpecificHeaders.putIfAbsent(TIMESTAMP, currentTimeMillis);
+    kafkaSpecificHeaders.putIfAbsent(MESSAGE_TIMESTAMP, currentTimeMillis);
+    return kafkaSpecificHeaders;
+  }
+
+  private Map<String, Object> prepareCustomHeaders(Map<String, Object> inputHeaders) {
+    Map<String, Object> customHeaders =
+        inputHeaders.entrySet().stream()
+            .filter(header -> !isKafkaSpecificHeader(header.getKey()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+    return serializeValuesAndAddKafkaSpecificHeader(customHeaders);
+  }
+
+  private boolean isKafkaSpecificHeader(String key) {
+    return kafkaSpecificHeaderKeys.contains(key);
+  }
+
+  private Map<String, Object> serializeValuesAndAddKafkaSpecificHeader(
+      Map<String, Object> inputCustomHeaders) {
```

Comment:
just a hint: you used _customHeaders_ up there for the same parameter in the same context. 
I feel it would be easier to follow if you stay consistent with the naming unless the variable is used in different context or have different purpose.